ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt

```diff
@@ -54,59 +55,85 @@ constructor(
-
-    return locationOfInterestMutations.combine(submissionMutations, this::combineAndSortMutations)
-  }
+  /**
+   * Return the set of data upload queue entries not yet marked as completed sorted in chronological
+   * order (FIFO). Media/photo uploads are not included.
+   */
+  suspend fun getIncompleteUploads(): List<UploadQueueEntry> =
+    getUploadQueueFlow().first().filter {
+      setOf(PENDING, IN_PROGRESS, FAILED, UNKNOWN).contains(it.uploadStatus)
+    }
 
   /**
-   * Returns all local submission mutations associated with the the given LOI ID that have one of
-   * the provided sync statues.
+   * Return the set of photo/media upload queue entries not yet marked as completed, sorted in
+   * chronological order (FIFO).
    */
-  suspend fun getSubmissionMutations(
-    loiId: String,
-    vararg entitySyncStatus: MutationEntitySyncStatus,
-  ) = getMutations(loiId, *entitySyncStatus).filterIsInstance<SubmissionMutation>()
+  suspend fun getIncompleteMediaMutations(): List<SubmissionMutation> =
+    getUploadQueueFlow()
+      .first()
+      .filter {
+        setOf(MEDIA_UPLOAD_PENDING, MEDIA_UPLOAD_IN_PROGRESS, MEDIA_UPLOAD_AWAITING_RETRY)
+          .contains(it.uploadStatus)
+      }
+      // TODO(https://github.com/google/ground-android/issues/2120):
+      //  Return [MediaMutations] instead once introduced.
+      .mapNotNull { it.submissionMutation }
 
   /**
-   * Returns all LOI and submission mutations in the local mutation queue relating to LOI with the
-   * specified id, sorted by creation timestamp (oldest first).
+   * Returns a [Flow] which emits the upload queue once and on each change, sorted in chronological
+   * order (FIFO).
    */
-  suspend fun getMutations(
-    loiId: String,
-    vararg entitySyncStatus: MutationEntitySyncStatus,
-  ): List<Mutation> {
-    val loiMutations =
-      localLocationOfInterestStore
-        .findByLocationOfInterestId(loiId, *entitySyncStatus)
-        .map(LocationOfInterestMutationEntity::toModelObject)
-    val submissionMutations =
-      localSubmissionStore.findByLocationOfInterestId(loiId, *entitySyncStatus).map {
-        it.toSubmissionMutation()
+  private fun getUploadQueueFlow(): Flow<List<UploadQueueEntry>> =
+    localLocationOfInterestStore.getAllMutationsFlow().combine(
+      localSubmissionStore.getAllMutationsFlow()
+    ) { loiMutations, submissionMutations ->
+      buildUploadQueue(loiMutations, submissionMutations)
+    }
+
+  private suspend fun buildUploadQueue(
+    loiMutations: List<LocationOfInterestMutation>,
+    submissionMutations: List<SubmissionMutation>,
+  ): List<UploadQueueEntry> {
+    val user = authenticationManager.getAuthenticatedUser()
+    val loiMutationMap = loiMutations.filterByUser(user).associateBy { it.collectionId }
+    val submissionMutationMap =
+      submissionMutations.filterByUser(user).associateBy { it.collectionId }
+    val collectionIds = loiMutationMap.keys + submissionMutationMap.keys
```

Comment:
Do we need to remove duplicates?

Reply:
`collectionIds` is a `Set, so no need. I've added an explicit type to the `val` to make this clear.